utils/google-document.js

```diff
@@ -462,7 +463,11 @@ class GoogleDocument {
 
       Object.keys(this.crosslinksPaths).forEach((id) => {
         elementsStringify = elementsStringify.replace(
-          `https://docs.google.com/document/d/${id}`,
+          new RegExp(
+            `https://docs\\.google\\.com/document/(?:u/\\d+/)?d\\/${escapeRegexp(
```

Comment:
Please don't add a dependency for that. 
Replace by `https:\/\/docs.google.com\/document\/d\/[a-zA-Z0-9]+(?:\/edit)?`

Reply:
what is the meaning of `(?:u/\\d+/)?`? Do you have some documents without `/d/`?

Reply:
Ok some urls has `userID`. [Source](https://support.google.com/a/answer/6175982?hl=en).
I also found urls can ends with `/preview`

new regex `https:\/\/docs.google.com\/document\/(?:u\/\d+\/)?d\/[a-zA-Z0-9]+(?:(?:\/edit)|(?:\/preview))?`

Reply:
the extra non-capturing groups aren't needed for `|`, so i'll use
`https:\/\/docs.google.com\/document\/(?:u\/\d+\/)?d\/[a-zA-Z0-9]+(?:\/edit|\/preview)?`

Reply:
> Please don't add a dependency for that.
> Replace by `https:\/\/docs.google.com\/document\/d\/${id}(?:\/edit)?`

Instead, i used the [function second parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter) which makes the code a bit shorter.